pkg/cli/connect.go

```diff
@@ -0,0 +1,189 @@
+		if err != nil {
+			return fmt.Errorf("get package node fail: %v", err)
+		}
+		err = connectToPackageNode(that, args[1:])
+		if err != nil {
+			return fmt.Errorf("connectToPackageNode fail: %v", err)
+		}
+	default:
+		return fmt.Errorf("unsuported node type %T", thatVal)
+	}
+	return nil
+}
+
+func connectToFileNode(node *service.FileNode, args []string) error {
+	for _, nID := range args {
+		thisID, err := ParseNodeID(nID)
+		if err != nil {
+			return fmt.Errorf("ParseNodeID fail for %q: %v", args[0], err)
+		}
+		switch thisVal := thisID.(type) {
+		// FileNode -> FileNode
+		case *service.FileNode:
+			this, err := getUniqueFileNode(thisVal)
+			if err != nil {
+				return fmt.Errorf("get unique \"this\" node fail: %v", err)
+			}
+			// Which edge
+			switch connectCmdFlags.fileToFileEdge {
+			case "derivedFrom":
+				node.DerivedFrom = append(node.DerivedFrom, this)
+			case "dependencies":
+				node.Dependencies = append(node.Dependencies, this)
+			default:
+				return fmt.Errorf("unknown edge for FileNode -> FileNode: %s", connectCmdFlags.fileToFileEdge)
+			}
+		default:
+			return fmt.Errorf("cannot connect %T to FileNode", thisVal)
+		}
+	}
+	// ship it back
+	err := sendFileNode(node)
+	if err != nil {
+		return fmt.Errorf("sending FileNode fail: %v", err)
+	}
+	return nil
+}
+
+func connectToPackageNode(node *service.PackageNode, args []string) error {
+	stream, err := buildServiceClient.Package(context.Background())
+	if err != nil {
+		return err
+	}
+	for _, nID := range args {
+		thisID, err := ParseNodeID(nID)
+		if err != nil {
+			return fmt.Errorf("ParseNodeID fail for %q: %v", args[0], err)
+		}
+		switch thisVal := thisID.(type) {
+		// FileNode -> FileNode
+		case *service.FileNode:
+			this, err := getUniqueFileNode(thisVal)
+			if err != nil {
+				return fmt.Errorf("get unique \"this\" node fail: %v", err)
+			}
+			// Which edge
+			switch connectCmdFlags.fileToPackageEdge {
+			case "targets":
+				err = stream.Send(this)
+				if err != nil {
+					return fmt.Errorf("send fileNode to pkg stream fail: %v", err)
+				}
+			default:
+				return fmt.Errorf("unknown edge for FileNode -> FileNode: %s", connectCmdFlags.fileToPackageEdge)
+			}
+		default:
+			return fmt.Errorf("cannot connect %T to FileNode", thisVal)
+		}
+	}
+	res, err := stream.CloseAndRecv()
+	if err != nil {
+		return fmt.Errorf("close stream fail: %v", err)
+	}
+	if !res.Success {
+		return fmt.Errorf("sending node fail: %v", err)
+	}
+	return nil
+}
+
+func getUniqueFileNode(queryNode *service.FileNode) (*service.FileNode, error) {
+	stream, err := controlServiceClient.GetFileNode(context.Background(), queryNode)
+	node, err := stream.Recv()
+	if err != nil {
+		return nil, fmt.Errorf("recv FileNode fail for %v: %v", queryNode, err)
+	}
+	_, err = stream.Recv()
+	if err == nil {
+		return nil, fmt.Errorf("more than one FileNode match %v", queryNode)
```

Comment:
Here I would also prompt the user to use a better identifier for the node

Reply:
How would you do that?

Reply:
Include it in the error message